Formula/catboost-cli.rb

```diff
@@ -0,0 +1,44 @@
+class CatboostCli < Formula
+  desc "Fast, scalable, high performance Gradient Boosting on Decision Trees cli tool"
+  homepage "https://catboost.ai"
+  url "https://github.com/catboost/catboost/archive/refs/tags/v1.2.1.tar.gz"
+  sha256 "b1cf4458415a639f26e953bc4e2877d68d2a711f457a725ef93cbe24363725f0"
+  license "Apache-2.0"
+  head "https://github.com/catboost/catboost.git", branch: "master"
+
+  livecheck do
+    url :stable
+    strategy :github_latest
+  end
+
+  depends_on "cmake" => :build
+  depends_on "conan@1" => :build
+  depends_on "ninja" => :build
+
+  on_linux do
+    depends_on "llvm" => :build
+  end
+
+  def install
+    extra_cmake_args = [
+      "-S",
+      buildpath.to_s,
+      "-B",
+      "#{buildpath}/brew-build",
+      "-G",
+      "Ninja",
+      "-DCATBOOST_COMPONENTS=app",
+      "-DHAVE_CUDA=NO",
+      "-DCMAKE_TOOLCHAIN_FILE=#{buildpath}/build/toolchains/clang.toolchain",
+    ]
+    system "cmake", *extra_cmake_args, *std_cmake_args
+    system "ninja", "-C", "#{buildpath}/brew-build", "catboost"
+
+    bin.install "#{buildpath}/brew-build/catboost/app/catboost" => "catboost-#{version}"
+    bin.install_symlink "catboost-#{version}" => "catboost"
+  end
+
+  test do
+    assert_predicate bin/"catboost", :exist?
```

Comment:
we need a better test

---

We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
I've managed to add a test that checks the installed binary is working as expected.